src/util/progress.ts

```diff
@@ -20,30 +22,34 @@ export class Progress {
         (progress: vscode.Progress<{increment: number, message: string}>, token: vscode.CancellationToken) => {
             const calls: (()=>Promise<any>)[] = [];
             steps.reduce((previous: Step, current: Step, currentIndex: number, steps: Step[])=> {
+                current.total = previous.total + current.increment;
                 calls.push(()=> {
-                    let result: Promise<any> = Promise.resolve();
-                    progress.report({
-                        increment: previous.increment,
-                        message: `${previous.total}%`
-                    });
-                    result = odo.execute(current.command);
-                    current.total = previous.total + current.increment;
-                    return currentIndex+1 === steps.length ? result.then(()=> {
+                    return new Promise<CliExitData>(async (resolve,reject) => {
                         progress.report({
                             increment: previous.increment,
                             message: `${previous.total}%`
                         });
-                    }) : result;
+
+                            const result: CliExitData = await odo.execute(current.command);
```

Comment:
If an error happens here, it gets swallowed and the progress is stalled. You might need to call this with fail = false to pass it into the handling below.